File: c++/e998.cpp
#include <iostream>
using namespace std;

int main(){
    int n;
    while (cin >> n){
        int num=1;
        for (int i = 0; i <n;i++){
            if (i%2){
                for (int j =n-1;j>-1;j--) cout << num +j << ' ';
            }else{
                for (int j=0;j<n;j++) cout << num+j << ' ';
            }
            num+=n;
            cout << "\n";
        }
        cout << '\n';
    }
}

File: python/e998.py
import sys

for n in sys.stdin:
    n=int(n)
    num=1
    for i in range(n) :
        if i%2:
            for j in range(n-1,-1,-1): print(num+j,end=' ')
        else: 
            for j in range(n): print(num+j,end=' ')
        num+=n
        print()
    print()
